test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -131,6 +153,59 @@ TEST_F(XdsOverrideHostTest, ConfigRequiresChildPolicy) {
             absl::InvalidArgumentError(
                 "errors validating xds_override_host LB policy config: "
                 "[field:childPolicy error:field not present]"));
+
+  result =
+      CoreConfiguration::Get().lb_policy_registry().ParseLoadBalancingConfig(
+          Json::Array{Json::Object{
+              {"xds_override_host_experimental",
+               Json::Object{
+                   {"childPolicy",
+                    Json::Array{
+                        {Json::Object{{"pick_first", Json::Object{}}}},
+                        {Json::Object{{"round_robin", Json::Object{}}}}}}}
+
+              }}});
+  EXPECT_EQ(result.status(),
+            absl::InvalidArgumentError(
+                "errors validating xds_override_host LB policy config: "
+                "[field:childPolicy error:exactly one child config should be "
+                "specified]"));
+
+  result =
+      CoreConfiguration::Get().lb_policy_registry().ParseLoadBalancingConfig(
+          Json::Array{
+              Json::Object{{"xds_override_host_experimental",
+                            Json::Object{{
+                                "childPolicy",
+                                {Json::Object{{"pick_first", Json::Object{}}}},
+                            }}}
+
+              }});
+  EXPECT_EQ(result.status(),
+            absl::InvalidArgumentError(
+                "errors validating xds_override_host LB policy config: "
+                "[field:childPolicy error:type should be array]"));
+
+  result =
+      CoreConfiguration::Get().lb_policy_registry().ParseLoadBalancingConfig(
+          Json::Array{Json::Object{{"xds_override_host_experimental",
+                                    Json::Object{{"childPolicy", Json::Array{}}}
+
+          }}});
+  EXPECT_EQ(result.status(),
+            absl::InvalidArgumentError(
+                "errors validating xds_override_host LB policy config: "
+                "[field:childPolicy error:No known policies in list: ]"));
+}
+
+TEST_F(XdsOverrideHostTest, XdsOverrideHostLbPolicyDisabled) {
+  policy_.reset();
```

Comment:
I suggest moving this to a separate test class, so that you don't need all of this boilerplate:

```
using XdsOverrideHostDisabledTest = LoadBalancingPolicyTest;

TEST_F(XdsOverrideHostDisabledTest, PolicyDoesNotExist) {
  EXPECT_EQ(MakeLbPolicy("xds_override_host_experimental"), nullptr);
}
```

(Note that this depends on my suggestion above to move `grpc_init()` and `grpc_shutdown()` back to `main()`.)